### ISSUE TITLE
[Bug] The definition of total violation is wrong.

### ISSUE THREAD

**Issue by Jiangyan-Zhao** [pytorch/botorch#2031]
# 🐛 Bug

In the modules below, the definition of **total violation** is wrong. 
It should be $\sum_{\ell=1}^m\max [c_{\ell}(x),0]$, not $\sum_{\ell=1}^m c_{\ell}(x)$.

- tutorial [scalable_constrained_bo.ipynb](https://github.com/pytorch/botorch/blob/main/tutorials/scalable_constrained_bo.ipynb) 
    - function `update_state()` 
- botorch.generation.sampling.ConstrainedMaxPosteriorSampling

## Example

Replace
```python
sum_violation = C_next.sum(dim=-1)
```
 in the function `update_state()` of the chunk [4] of the tutorial [scalable_constrained_bo.ipynb](https://github.com/pytorch/botorch/blob/main/tutorials/scalable_constrained_bo.ipynb) with
```python
violation = torch.where(C_next > 0, C_next, 0)
sum_violation = violation.sum(dim=-1)
```

**Comment by saitcakmak** [comment]
Thanks for reporting! cc @nataliemaus

**Comment by dme65** [comment]
Fixed in #2034.
